cloudinit/sources/DataSourceOVF.py

```diff
@@ -129,26 +129,82 @@ def _get_data(self):
                 else:
                     LOG.debug("Did not find the customization plugin.")
 
+                metaPath = None
                 if vmwareImcConfigFilePath:
+                    imcdirpath = os.path.dirname(vmwareImcConfigFilePath)
+                    cf = ConfigFile(vmwareImcConfigFilePath)
+                    self._vmware_cust_conf = Config(cf)
                     LOG.debug("Found VMware Customization Config File at %s",
                               vmwareImcConfigFilePath)
-                    nicspath = wait_for_imc_cfg_file(
-                        filename="nics.txt", maxwait=10, naplen=5)
+                    try:
+                        (metaPath, userPath, nicspath) = collect_imc_files(
+                            self._vmware_cust_conf)
+                    except Exception as e:
+                        _raise_error_status(
+                            "File(s) missing in directory",
+                            e,
+                            GuestCustEvent.GUESTCUST_EVENT_CUSTOMIZE_FAILED,
+                            vmwareImcConfigFilePath,
+                            self._vmware_cust_conf)
                 else:
                     LOG.debug("Did not find VMware Customization Config File")
-            else:
-                LOG.debug("Customization for VMware platform is disabled.")
 
-        if vmwareImcConfigFilePath:
+                # ignore disable_vmware_customization if meta data is available
+                if not metaPath:
+                    if util.get_cfg_option_bool(self.sys_cfg,
+                                                "disable_vmware_customization",
+                                                True):
+                        LOG.debug(
+                            "Customization for VMware platform is disabled.")
+                        # reset vmwareImcConfigFilePath to None to avoid
+                        # customization for VMware platform
+                        vmwareImcConfigFilePath = None
+
+        if vmwareImcConfigFilePath and metaPath:
+            try:
+                set_gc_status(self._vmware_cust_conf, "Started")
+
+                LOG.debug("Start to load cloud-init meta data and user data")
+                (md, ud, cfg, network) = load_cloudinit_data(
+                    metaPath, userPath)
+
+                if network:
+                    self._network_config = network
+                else:
+                    fallbackNetwork = self.distro.generate_fallback_config()
+                    self._network_config = fallbackNetwork
+
+            except safeyaml.YAMLError as e:
```

Comment:
A couple of minor points here:
 - generally it's nice to only include operations within the try/except blocks that are likely to generate the exceptions we are handling. So we should move things like LOG.deb and set_gc_status outside of the try. It makes it easier to understand what callsites are likely to raise errors
 -  I don't think we need to create and assign a fallbackNetwork local variable as we aren't using that anywhere else

The changeset I was thinking is here

```suggestion
            LOG.debug("Start to load cloud-init meta data and user data")
            set_gc_status(self._vmware_cust_conf, "Started")
            try:
                (md, ud, cfg, network) = load_cloudinit_data(md_path, ud_path)
                if network:
                    self._network_config = network
                else:
                    self._network_config = (
                        self.distro.generate_fallback_config()
                    )
            except safeyaml.YAMLError as e:
```

Reply:
Fixed. Your points 2) is quite useful. I created a local var because I couldn't write "self._network_config = self.distro.generate_fallback_config" in one line. From your comments, I learnt that we could use () to split it into 2 lines. Thanks a lot.